pkg/image/apis/image/types.go

```diff
@@ -469,6 +469,36 @@ type ImageStreamImage struct {
 // DockerImageReference points to a Docker image.
 type DockerImageReference = reference.DockerImageReference
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImageStreamLayers describes information about the layers referenced by images in this
+// image stream.
+type ImageStreamLayers struct {
+	metav1.TypeMeta
+	// Standard object's metadata.
+	metav1.ObjectMeta
+	// blobs is a map of blob name to metadata about the blob.
+	Blobs map[string]ImageLayerData
+	// layers is a map between an image name and the names of the blobs that form the
+	// images.
+	Layers map[string]BlobReferences
+	// manifests is a map between an image name and the name of the manifest blob, if one
```

Comment:
parallel maps with the same key is weird.  How about one map with a struct containing the values you want.